src/integration-test/java/com/commercetools/emailprocessor/integration/commons/jobs/EmailJobIT.java

```diff
@@ -88,12 +88,28 @@ public void process_withASuccessfulEmail_shouldReturnNoError() {
         assertThat(statistics).isNotEmpty();
         Statistics statistic = statistics.get(0);
         assertThat(statistic.getProcessed()).isEqualTo(2);
-        assertThat(statistic.getNotProcessed()).isEqualTo(1);
         assertThat(statistic.getSentSuccessfully()).isEqualTo(2);
         assertThat(statistic.getTemporaryErrors()).isEqualTo(0);
         assertThat(statistic.getPermanentErrors()).isEqualTo(0);
     }
 
+    @Test
+    public void process_withManyEmailsObjects_shouldReturnNoError() {
+        int customObjectId = 0;
+        while (customObjectId < 100) {
+            createCustomObject(EmailProcessor.STATUS_PENDING, String.valueOf(customObjectId));
+            customObjectId = customObjectId + 1;
+        }
+        configuration.getTenants().get(0).setEndpointUrl(HTTPBIN_DOMAIN + Statistics.RESPONSE_CODE_SUCCESS);
+        List<Statistics> statistics = EmailJob.process(configuration).toCompletableFuture().join();
+        assertThat(statistics).isNotEmpty();
+        Statistics statistic = statistics.get(0);
+        assertThat(statistic.getProcessed()).isEqualTo(100);
+        assertThat(statistic.getSentSuccessfully()).isEqualTo(100);
+        assertThat(statistic.getTemporaryErrors()).isEqualTo(0);
+        assertThat(statistic.getPermanentErrors()).isEqualTo(0);
+    }
```

Comment:
Maybe it makes sense to add ITs for asserting that the email objects have all been processed after running the process (that's in the case if processAll for example) and one for the case (when not processAll is set and there are errored emails for example). So the cases to test are:

`process_withOnlyPendingEmailObjectsAndProcessAllIsFalse_shouldProcessAll`
`process_withOnlyPendingEmailObjectsAndProcessAllIsTrue_shouldProcessAll`
`process_withMixedStateEmailObjectsAndProcessAllIsFalse_shouldNotProcessAll`
`process_withMixedStateEmailObjectsAndProcessAllIsTrue_shouldProcessAll`

These ITs will assert that the feature that you added works now

Reply:
879b5c3